src/pydata_sphinx_theme/theme/pydata_sphinx_theme/layout.html

```diff
@@ -1,3 +1,9 @@
+{# We redefine <html/> for "basic/layout.html" to add a default `data-theme` attribute when
+ # a default mode has been set. This also improves compatibility when JavaScript is disabled.
+ #}
+{% set html_tag %}
+<html{% if not html5_doctype %} xmlns="http://www.w3.org/1999/xhtml"{% endif %}{% if language is not none %} lang="{{ language }}"{% endif %} {% if default_mode %}data-theme="{{ default_mode }}"{% endif %}>
```

Comment:
`default_mode` is set by the documentation provider. If the project homepage has a black background, I would set `default_mode` to dark so that random visitors don’t get a huge contrast shift when clicking on the “docs” link on the homepage.

`data-theme` can be set by users and is saved in the localStorage. The default value written in the HTML file will be overridden if that’s the case (and they have JavaScript enabled).